File: src/models/flight.ts
import ConfirmationNumber from "./confirmation"; 

interface Flight {
    Id: number;
    FlightNumber: number;
    DepartureDate: string;
    ArrivalDate: string;
    DepartureTime: string;
    ArrivalTime: string;
    DepartureAirport: string;
    ArrivalAirport: string; 
    PassengerLimit: number; 
    Bookings: ConfirmationNumber[]; 
}

export default Flight;

File: SMAirlines/Models/Flight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.Json.Serialization;

namespace SMAirlines.Models
{
    public class Flight
    {
        public int Id { get; set; }
        public int FlightNumber { get; set; }
        public DateTime DepartureDate { get; set; } = new DateTime(); 
        public DateTime ArrivalDate { get; set; } = new DateTime();
        public DateTime DepartureTime { get; set; } = new DateTime();
        public DateTime ArrivalTime { get; set; } = new DateTime();
        public string DepartureAirport { get; set; } = String.Empty;
        public string ArrivalAirport { get; set; } = String.Empty;
        public int PassengerLimit { get; set; } = 0;
        public List<ConfirmationNumber> Bookings { get; set; } = null!; 
    }
}
